test/scheduler_test.js

```diff
@@ -19,6 +19,7 @@ suite('Scheduler', function() {
       Hook: helper.Hook,
       taskcreator: creator,
       pollingDelay: 1,
+      notify: helper.notify,
```

Comment:
Ah, this is a *much* easier kind of dependency injection than I was suggesting.  You can ignore all the stuff I said about loader and overwrites :)

Since the fake notify is only used here, you could just require fakeNotify in this file, similar to how creator is set up a few lines earlier.